File: .claude/skills/sast-scan/rules/semgrep/ruby/tests/test_cwe79_xss.rb
# CWE-79: XSS / CWE-601: Open Redirect
def test_xss(user_input)
  # ruleid: ruby.security.xss-raw
  raw(user_input)
  # ruleid: ruby.security.xss-raw
  user_input.html_safe

  # ok: ruby.security.xss-raw
  raw(h(user_input))
  # ok: ruby.security.xss-raw
  h(user_input)
end

File: .claude/skills/sast-scan/rules/semgrep/php/tests/test_cwe79_xss.php
<?php
// CWE-79: Cross-Site Scripting

function testXss(string $name, string $msg) {
    // ruleid: php.security.xss-echo-unescaped
    print $name;
    // ruleid: php.security.xss-echo-unescaped
    printf("Hello %s", $msg);

    // ok: php.security.xss-echo-unescaped
    print htmlspecialchars($name);
    // ok: php.security.xss-echo-unescaped
    echo "Hello World";
}
